src/chain/chain.rs

```diff
@@ -307,6 +308,17 @@ impl Chain {
         Ok(vec![self.state.new_info.clone()])
     }
 
+    /// Starts pruning as requested. Returns a new `SectionInfo`.
+    pub fn prune(&mut self) -> Result<SectionInfo, RoutingError> {
+        if self.state.change != PrefixChange::None {
+            log_or_panic!(LogLevel::Warn, "Tring to prune during pfx change.",);
+        }
+
+        self.state.change = PrefixChange::Pruning;
+
+        Ok(self.state.new_info.clone())
```

Comment:
Why is prune returning a sectionInfo?

Reply:
Yeah, an empty return shall be enough, I will correct it.